server/server.js

```diff
@@ -25,551 +27,518 @@ const unrespondedSessionAlertTimeoutMillis = helpers.getEnvVar('FALLBACK_TIMEOUT
+        currentSession.incrementButtonPresses(numPresses)
+        currentSession.updateBatteryLevel(batteryLevel)
+        await db.saveSession(currentSession, client)
+      }
+    } else if (currentSession === null) {
+      currentSession = await db.createSession(button.installation_id, button.button_id, button.unit, button.phone_number, numPresses, null, client)
+    } else {
+      currentSession.incrementButtonPresses(numPresses)
+      await db.saveSession(currentSession, client)
     }
-}
 
-function needToSendButtonPressMessageForSession(session) {
-    return (session.numPresses === 1 || session.numPresses === 2 || session.numPresses % 5 === 0)
-}
+    if (needToSendButtonPressMessageForSession(currentSession)) {
+      sendButtonPressMessageForSession(currentSession, client)
+    }
 
-async function sendButtonPressMessageForSession(session, client) {
+    await db.commitTransaction(client)
+  } catch (e) {
     try {
-        let installation = await db.getInstallationWithInstallationId(session.installationId, client)
-
-        if (session.numPresses === 1) {
-            const alertInfo = {
-                sessionId: session.id,
-                toPhoneNumber: installation.responderPhoneNumber,
-                fromPhoneNumber: session.phoneNumber,
-                message: `There has been a request for help from Unit ${session.unit.toString()} . Please respond "Ok" when you have followed up on the call.`,
-                reminderTimeoutMillis: unrespondedSessionReminderTimeoutMillis,
-                fallbackTimeoutMillis: unrespondedSessionAlertTimeoutMillis,
-                reminderMessage: 'Please Respond "Ok" if you have followed up on your call. If you do not respond within 2 minutes an emergency alert will be issued to staff.',
-                fallbackMessage: `There has been an unresponded request at ${installation.name} unit ${session.unit.toString()}`,
-                fallbackToPhoneNumber: installation.fallbackPhoneNumber,
-                fallbackFromPhoneNumber: helpers.getEnvVar('TWILIO_FALLBACK_FROM_NUMBER'),
-            }
-            braveAlerter.startAlertSession(alertInfo)
-        }
-        else if (session.numPresses % 5 === 0 || session.numPresses === 2) {
-            braveAlerter.sendSingleAlert(
-                installation.responderPhoneNumber,
-                session.phoneNumber,
-                `This in an urgent request. The button has been pressed ${session.numPresses.toString()} times. Please respond "Ok" when you have followed up on the call.`,
-            )
-        }
-    }
-    catch (e) {
-        helpers.log(`sendButtonPressMessageForSession: Failed to start alert session for ${session.phoneNumber}: ${JSON.stringify(e)}`)
+      await db.rollbackTransaction(client)
+      helpers.log(`handleValidRequest: Rolled back transaction because of error: ${e}`)
+    } catch (error) {
+      // Do nothing
+      helpers.log(`handleValidRequest: Error rolling back transaction: ${e}`)
```

Comment:
This is unrelated to this pull request, but @trisapeace it seems like the log message is referring to one error but logging the other one - it might make sense to log both here.

Reply:
Yes, you're right. That should definitely be logging `error` on line 126. I like you're idea of logging both.

@risatronic do you mind updating this to
```
      // Do nothing
      helpers.log(`handleValidRequest: Error rolling back transaction: ${error} Rollback attempted because of error: ${e}`)
```
?

Thanks!

Reply:
done :) I think this is good to be merged now because it is running ok on dev?